### PR TITLE
Added __file__ and __name__ to xonsh scripts

### PR DESCRIPTION
**Pull request by scopatz**
This should address #2178

### CODE REVIEW

**Comment by scopatz** [comment]
Ok, the tests finally pass CI here

File: tests/tools.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 ON_WINDOWS = (platform.system() == 'Windows')
 ON_CONDA = True in [conda in pytest.__file__ for conda
                     in ['anaconda', 'miniconda']]
+ON_TRAVIS = 'TRAVIS' is os.environ and 'CI' in os.environ
```
Comment (laerus) [inline tests/tools.py]: `'TRAVIS' in* os.environ` ?

File: xonsh/tools.py [hunk 2]
```diff
@@ -805,6 +805,21 @@ def swap(namespace, name, value, default=NotImplemented):
     else:
         setattr(namespace, name, old)
 
+
+@contextlib.contextmanager
+def swap_values(d, updates, default=NotImplemented):
+    """Updates a dictionary (or other mapping) with values from another mapping,
+    and then restores the original mapping when the context is exited.
+    """
+    old = {k: d.get(k, NotImplemented) for k in updates}
```
Comment (laerus) [inline xonsh/tools.py]: `NotImplemented` can be the value of an existing key in the old dict,
isn't an instance of `object` the suggested sentinel value?

File: tests/tools.py (hunk 1 above)
Comment (scopatz) [inline tests/tools.py]: Oh shoot! Good catch @laerus!

File: xonsh/tools.py (hunk 2 above)
Comment (scopatz) [inline xonsh/tools.py]: That is a good idea.  Where did you see that suggested, though? Just out of curiosity...

**Comment by scopatz** [comment]
Thanks for the review @laerus, I think that this is ready for another look!

File: xonsh/tools.py (hunk 2 above)
Comment (laerus) [inline xonsh/tools.py]: picked that up from a python core dev talk (iirc Raymond Hettinger on idiomatic python)
and kind of took it for granted, no idea if there is anything on the official docs.

**Comment by laerus** [comment]
thanks @scopatz !

File: xonsh/tools.py (hunk 2 above)
Comment (scopatz) [inline xonsh/tools.py]: Nice, it is definitely the way to do it.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**